lab-brian/src/reducer/category.js

```diff
@@ -0,0 +1,22 @@
+'use strict';
+
+let initialState = [];
+
+export default(state=initialState, action) => {
+  let { type, payload } = action;
+  // same as type = action.type, payload = action.payload
+  switch(type) {
+    case 'CATEGORY_CREATE':
+      return [...state, payload];
+    case 'CATEGORY_UPDATE':
+      return state.map(category => 
+        category.id === payload.id ? payload : category);
+    case 'CATEGORY_DELETE':
+      return state.filter(category =>
+        category.id !== category.id);
```

Comment:
When you click delete, everything gets removed instead of just the one thing. My guess is that category doesn't have the id you think it has, similar to the problem that judy was having in class today.